route/gin_group.go

```diff
@@ -111,7 +111,6 @@ func (r *GinGroup) getGinRoutesWithMiddlewares() gin.IRoutes {
 	middlewares = append(middlewares, ginOriginMiddlewares...)
 	middlewares = append(middlewares, ginMiddlewares...)
 	middlewares = append(middlewares, ginLastMiddlewares...)
-	r.middlewares = []httpcontract.Middleware{}
```

Comment:
So sorry, just saw this PR! If we remove this line directly, maybe we will get the error below:
```
facades.Route.Middleware(A).Get()
facades.Route.Middleware(B).Post() // will apply middleware A and B at the same time
```

We can add a test case to confirm this.

Reply:
> So sorry, just saw this PR! If we remove this line directly, maybe we will get the error below:
> 
> ```
> facades.Route.Middleware(A).Get()
> facades.Route.Middleware(B).Post() // will apply middleware A and B at the same time
> ```
> 
> We can add a test case to confirm this.

Any suggestions for improvement? I'm not very familiar with Gin's middleware usage.

Reply:
Maybe we can move this line to a new method: `func (r *GinGroup) clearMiddlewares()`, then call the method at the end of methods of calling `r.getGinRoutesWithMiddlewares()`: `Get`, `Post`, etc.
```
func (r *GinGroup) Any(relativePath string, handler httpcontract.HandlerFunc) {
	r.getGinRoutesWithMiddlewares().Any(pathToGinPath(relativePath), []gin.HandlerFunc{handlerToGinHandler(handler)}...)
	r.clearMiddlewares()
}

func (r *GinGroup) Resource(relativePath string, controller httpcontract.ResourceController) {
	r.getGinRoutesWithMiddlewares().GET(pathToGinPath(relativePath), []gin.HandlerFunc{handlerToGinHandler(controller.Index)}...)
	r.getGinRoutesWithMiddlewares().POST(pathToGinPath(relativePath), []gin.HandlerFunc{handlerToGinHandler(controller.Store)}...)
	r.getGinRoutesWithMiddlewares().GET(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Show)}...)
	r.getGinRoutesWithMiddlewares().PUT(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Update)}...)
	r.getGinRoutesWithMiddlewares().PATCH(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Update)}...)
	r.getGinRoutesWithMiddlewares().DELETE(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Destroy)}...)
	r.clearMiddlewares()
}
```

Reply:
ok, fixed it.

Reply:
Great! Could you please add a test case to cover this situation?
```
facades.Route.Middleware(A).Get()
facades.Route.Middleware(B).Post() // will apply middleware A and B at the same time
```